2. Создание Опросника

VitaliyIliyushenkov opened the pull request.

VitaliyIliyushenkov committed:
Add

VitaliyIliyushenkov committed:
Update Form1.cs

Упрощенная версия

VitaliyIliyushenkov committed:
Упрощенная версия

VasilevsWork commented on Survey_Test/Survey_Test/Form1.cs:
```diff
@@ -0,0 +1,190 @@
+        private void Save_file_btn_Click(object sender, EventArgs e)
+        {
+            OpenFileDialog od = new OpenFileDialog();
+            if (od.ShowDialog() == DialogResult.OK)
```
По идее пользователю не нужно давать возможность указывать путь к файлу. Это встроенная возможность, но не принципиально.

VasilevsWork commented on Survey_Test/Survey_Test/Form1.cs:
```diff
@@ -0,0 +1,190 @@
+            {
+                path = od.FileName;
+            }
+            File.WriteAllText(path, "");
```
Произойдет падение, если закрыть диалог выбора файла (на крестик), не выбрав файл. Нужно убеждаться, что такой файл существует.

VasilevsWork commented on Survey_Test/Survey_Test/TestResult.cs:
```diff
@@ -0,0 +1,43 @@
+            String.Join(",", answer2) + '/' + answer3 + '/' + answer4 + '/' + answer5;
+            File.AppendAllText(path, createText + "\n");
+        }
+        public void Deserialize(string path, string CREATETEXT)
```
CREATETEXT немного странное название. SerializedData?

VasilevsWork commented on Survey_Test/Survey_Test/Form1.cs:
```diff
@@ -0,0 +1,190 @@
+            }
+            
+
+            for (int i = 0; i < File.ReadAllLines(path).Length; i++)
```
Произойдет падение, если закрыть диалог выбора файла (на крестик), не выбрав файл.

VasilevsWork commented on Survey_Test/Survey_Test/Form1.cs:
```diff
@@ -0,0 +1,190 @@
+
+        private void Save_btn_Click(object sender, EventArgs e)
+        {
+            CreateResult();
```
Крайне желательно сделать проверку на заполненность полей. Иначе добавляются пустые записи, что некорректно. 
![image](https://github.com/VitaliyIliyushenkov/KP_1255_Iliushenkov_Vitaliy/assets/47122530/ec3a6f79-4299-4910-a059-3f58dbe1f132)

VasilevsWork reviewed (commented):
Реализация в целом устраивает. Нужно исправить небольшие замечания (падения при открытии файла и добавить проверку на пустоту) и можно будет заливать реквест.

VitaliyIliyushenkov committed:
Update Form1.cs

Теперь при пустых полях вы не сможете записать данные. Добавил проверку на неопределенность файла

VasilevsWork reviewed (commented):
Задание принимается. Можете заливать реквест.
